Ktisis/Interface/Components/BoneTree.cs

```diff
@@ -32,30 +49,49 @@ public class BoneTree {
 				}
 			}
 		}
+
+		private static bool BoneDoesMatch(Bone bone) => 
+			SearchText != "" && (new [] { bone.UniqueName, bone.LocaleName }).Any(
+				name => name.Contains(SearchText, StringComparison.CurrentCultureIgnoreCase)
+			);
+
 		private static void DrawBoneTreeNode(Bone bone) {
 			if (bone == null) return;
 
 			var children = bone.GetChildren();
+			var decendents = bone.GetDescendants();
+
+			bool hasChildInQuery = decendents.Any(BoneDoesMatch);
+			bool hasChildSelected = decendents.Any(Skeleton.IsBoneSelected);
+			bool isSelected = Skeleton.IsBoneSelected(bone);
+			bool isQueried = BoneDoesMatch(bone);
 
-			var flag = children.Count > 0 ? ImGuiTreeNodeFlags.OpenOnArrow : ImGuiTreeNodeFlags.Leaf;
-			if (Skeleton.IsBoneSelected(bone))
-				flag |= ImGuiTreeNodeFlags.Selected;
+			var criteria = HighlightCriteria.None;
 
-			var show = DrawBoneNode(bone, flag, () => OverlayWindow.SetGizmoOwner(bone.UniqueName));
-			if (show) {
+			if (isSelected) criteria ^= HighlightCriteria.Selected;
+			if (isQueried) criteria ^= HighlightCriteria.Queried;
+			if (hasChildSelected) criteria ^= HighlightCriteria.ChildSelected;
+			if (hasChildInQuery) criteria ^= HighlightCriteria.ChildQueried;
```

Comment:
Attempts to unify this into a single method, went well initially - but hit a wall with [CS0315](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0315) when the generic was not satisfied with using the enum (the XOR operator wasn't satisfied either, which would have required each value of the enum to be type cast as an integer).
```cs
var criteria = ApplyMulti(
  HighlightCriteria.None,
  new [] {
    (HighlightCriteria.Selected, Skeleton.IsBoneSelected(bone)),
    (HighlightCriteria.Queried, BoneDoesMatch(bone)),
    (HighlightCriteria.ChildSelected, decendents.Any(BoneDoesMatch)),
    (HighlightCriteria.ChildQueried, decendents.Any(bone => Skeleton.IsBoneSelected(bone)))
  }
);

private static T ApplyMulti<T>(T initial, (T, bool)[] pairs) where T : IBitwiseOperators<T, T, T> {
  T result = initial;

  foreach (var (value, condition) in pairs) {
    if (condition) result ^= value;
  }

  return result;
}
```
The use of an array only for this, would also have been excessive for its repeated calls, overall performance and impact on garbage collection.